crates/sui-authority-aggregation/src/lib.rs

```diff
@@ -66,33 +69,77 @@ where
         })
         .collect();
 
-    let mut current_timeout = initial_timeout;
-    let mut accumulated_state = initial_state;
-    // Then, as results become available fold them into the state using FReduce.
-    while let Ok(Some((authority_name, result))) = timeout(current_timeout, responses.next()).await
-    {
-        let authority_weight = committee.weight(&authority_name);
-        accumulated_state =
-            match reduce_result(accumulated_state, authority_name, authority_weight, result).await {
-                // In the first two cases we are told to continue the iteration.
-                ReduceOutput::Continue(state) => state,
-                ReduceOutput::ContinueWithTimeout(state, duration) => {
-                    // Adjust the waiting timeout.
-                    current_timeout = duration;
-                    state
-                }
-                ReduceOutput::Failed(state) => {
-                    return Err(state);
-                }
-                ReduceOutput::Success(result) => {
-                    // The reducer tells us that we have the result needed. Just return it.
-                    return Ok((result, responses));
-                }
+    match min_timeout {
+        Some(min_timeout) => {
+            let mut accumulated_state = initial_state;
+            let mut authority_to_result: HashMap<K, Result<V, E>> = HashMap::new();
+
+            while let Ok(Some((authority_name, result))) =
+                timeout(min_timeout, responses.next()).await
```

Comment:
I think this is not quite right, cuz when `min_timeout == 1.5s`, it means this sig aggregation loop immediately returns when the next available sig comes after 1.5s. For example, if the bridge itself has a bad network, the closest node will take 2s to respond, then it will get 0 sigs and return.